### ISSUE TITLE
Automatic simplification fails with singletons in special cases

### ISSUE THREAD

**Issue by TimBergner** [AlexandreDecan/portion#38]
The automatic simplification seems to fail with singletons if it closes exactly one gap. In this case the singleton is only added to the left open interval on the right of the sigleton.
Examples:
```python
P.open(1, 2) | P.open(2, 3) | P.singleton(2)
>>> (1,2) | [2,3)
P.open(1, 3) & ~P.singleton(2) | P.singleton(2)
>>> (1,2) | [2,3)
```
If the result is stored in a Variable `x`, further calculations might be faulty:
```python
P.open(1, 3) in x
>>> False
```

**Comment by TimBergner** [comment]
Of course one of the most important information was only implicit in the examples: the problem concerns the merge function.

**Comment by AlexandreDecan** [comment]
Hello, 

Thank you for having reported this issue! I'll have a look asap and try to release a hotfix asap as well :-)

**Comment by AlexandreDecan** [comment]
I found the issue. When intervals are simplified, the underlying atomic intervals are sorted by their lower bound. and an attempt to merge them is made. Since (2, 3) and [2] have the same lower bound, the order in which they are provided defines the order in which they will be considered (and that's why (1, 2) | [2] | (2, 3) works): it tries first to merge (1, 2) and (2, 3) (they can't), then (2, 3) and [2] (which results in [2, 3)). 

I'll confirm this asap, fix it, and release a hotfix.

**Comment by AlexandreDecan** [comment]
Fixed in 39e25646947cdd6a3351dc7c012ca0badcb829d2

Fix will be released in a few minutes as part of version 2.1.2.
